src/Symfony/Component/Console/Question/ChoiceQuestion.php

```diff
@@ -26,9 +26,9 @@ class ChoiceQuestion extends Question
     private string $errorMessage = 'Value "%s" is invalid';
 
     /**
-     * @param string                     $question The question to ask to the user
-     * @param array                      $choices  The list of available choices
-     * @param string|bool|int|float|null $default  The default answer to return
+     * @param string                      $question The question to ask to the user
+     * @param list<string|bool|int|float> $choices  The list of available choices
```

Comment:
`getDefaultValidator` has some logic determining whether choices are an associative array, so using `list<>` is wrong. Please use `array<>` instead to match the actual behavior.

And please also add the phpdoc for the `getChoices` method to return the same type